File: src/main/java/gregtechlite/gtlitecore/core/advancement/criterion/BasicCriterion.java
package gregtechlite.gtlitecore.core.advancement.criterion;

import gregtechlite.gtlitecore.api.advancement.impl.AbstractCriterion;
import net.minecraft.entity.player.EntityPlayerMP;

/**
 * Will always succeed when an {@code AdvancementTrigger} with this {@code AdvancementCriterion} is fired.
 */
public class BasicCriterion extends AbstractCriterion
{

    @Override
    public boolean test(EntityPlayerMP player)
    {
        return true;
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/core/advancement/criterion/BasicCriterion.kt
package gregtechlite.gtlitecore.core.advancement.criterion

import gregtechlite.gtlitecore.api.advancement.impl.AbstractCriterion
import net.minecraft.entity.player.EntityPlayerMP

/**
 * Will always succeed when an `AdvancementTrigger` with this `AdvancementCriterion` is fired.
 */
class BasicCriterion : AbstractCriterion()
{

    override fun test(player: EntityPlayerMP?): Boolean = true
}